exp/controllers/awsmanagedmachinepool_controller.go

```diff
@@ -182,6 +201,21 @@ func (r *AWSManagedMachinePoolReconciler) reconcileNormal(
 
 	ekssvc := eks.NewNodegroupService(machinePoolScope)
 
+	if machinePoolScope.ManagedMachinePool.Spec.AWSLaunchTemplate != nil {
+		if err := r.reconcileLaunchTemplate(machinePoolScope, ec2Scope); err != nil {
+			r.Recorder.Eventf(machinePoolScope.ManagedMachinePool, corev1.EventTypeWarning, "FailedLaunchTemplateReconcile", "Failed to reconcile launch template: %v", err)
+			machinePoolScope.Error(err, "failed to reconcile launch template")
+			return ctrl.Result{}, err
+		}
+
+		if err := r.reconcileTags(machinePoolScope, ec2Scope); err != nil {
+			return ctrl.Result{}, errors.Wrap(err, "error updating tags")
+		}
+
+		// set the LaunchTemplateReady condition
+		conditions.MarkTrue(machinePoolScope.ManagedMachinePool, expinfrav1.LaunchTemplateReadyCondition)
```

Comment:
We should probably mark this condition false in the above error scenarios, along with a reason.

Reply:
That makes sense. Added logic to mark the condition to false in this case.